phpunit.xml

```diff
@@ -1,27 +1,39 @@
 <?xml version="1.0"?>
-<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="vendor/phpunit/phpunit/phpunit.xsd" bootstrap="tests/bootstrap.php" colors="true" executionOrder="defects" cacheDirectory=".phpunit.cache">
+<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:noNamespaceSchemaLocation="vendor/phpunit/phpunit/phpunit.xsd" bootstrap="tests/bootstrap.php"
+         colors="true" executionOrder="defects" cacheDirectory=".phpunit.cache">
 
-  <testsuites>
-    <testsuite name="main">
-      <directory>tests</directory>
-      <directory>rules-tests</directory>
-      <directory>packages-tests</directory>
-      <directory>utils-tests</directory>
-    </testsuite>
-    <!-- divide the rules-tests roughly in 2 equal sized groups -->
-    <testsuite name="php-rules">
-        <directory>rules-tests/Php*</directory>
-    </testsuite>
-    <testsuite name="other-rules">
-        <directory>rules-tests/</directory>
-        <exclude>rules-tests/Php*</exclude>
-    </testsuite>
-  </testsuites>
-  <php>
-    <ini name="memory_limit" value="-1"/>
-  </php>
+    <testsuites>
+        <testsuite name="main">
+            <directory>tests</directory>
+            <directory>rules-tests</directory>
+            <directory>packages-tests</directory>
+            <directory>utils-tests</directory>
+        </testsuite>
+        <testsuite name="tests">
+            <directory>tests</directory>
+        </testsuite>
+        <!-- divide the rules-tests roughly in 2 equal sized groups -->
+        <testsuite name="php-rules">
+            <directory>rules-tests/Php*</directory>
+        </testsuite>
+        <testsuite name="other-rules">
+            <directory>rules-tests/</directory>
+            <exclude>rules-tests/Php*</exclude>
+        </testsuite>
+        <testsuite name="packages-tests">
```

Comment:
I don't see why we need testsuites when they just map 1:1 to filesystem paths

Reply:
I think as you introduced the use of testsuites, we should be consistent in the way to run tests, be testsuite-based instead of folder-based.
This avoid the mix of concept like in the previous matrix with some values being a path, some being a command line partial, while the matrix variable is named `path`

Reply:
This makes the config hard to read. Let's keep the complexity lower.